### PR TITLE
Couchbase workload with respinning ceph pods

### PR DESCRIPTION
**Pull request by shreekarSS**

### CODE REVIEW

File: ocs_ci/ocs/couchbase.py [hunk 1]
```diff
@@ -203,23 +208,48 @@ def create_couchbase_worker(self, replicas=1):
...
+
+        Returns:
+            pod status
+
+        """
+        app_pod_list = get_pod_name_by_pattern('cb-example', constants.COUCHBASE_OPERATOR)
+        app_pod = app_pod_list[random.randint(0, len(app_pod_list) - 1)]
+        logging.info(f"respin pod {app_pod}")
+        app_pod = get_pod_obj(app_pod, namespace=constants.COUCHBASE_OPERATOR)
+        app_pod.delete(wait=True, force=False)
```
Comment (OdedViner) [inline ocs_ci/ocs/couchbase.py:249]: Can change "app_pod" var name to "app_pod_obj" [line 248]?

File: tests/e2e/workloads/couchbase/test_couchbase_pod_respin.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    Deploy an CouchBase workload using operator
+    """
+    @pytest.fixture()
+    def cb_setup(self, couchbase_factory_fixture):
+        """
+        Creates couchbase workload
+        """
+        self.cb = couchbase_factory_fixture(
+            replicas=3, run_in_bg=True, skip_analyze=True)
+
```
Comment (OdedViner) [inline tests/e2e/workloads/couchbase/test_couchbase_pod_respin.py:24]: Need to change (pep8)
        self.cb = couchbase_factory_fixture(
            replicas=3, run_in_bg=True, skip_analyze=True
)

File: tests/e2e/workloads/couchbase/test_couchbase_pod_respin.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        if pod_name == 'couchbase':
+            self.cb.respin_couchbase_app_pod()
+        else:
+            disruption = Disruptions()
+            disruption.set_resource(resource=f'{pod_name}')
+            disruption.delete_resource()
+
+        while not (self.cb.result.done()):
+            import time
+            time.sleep(10)
```
Comment (OdedViner) [inline tests/e2e/workloads/couchbase/test_couchbase_pod_respin.py:54]: move "import time" to line number 3

File: ocs_ci/ocs/couchbase.py [hunk 4]
```diff
@@ -203,23 +208,48 @@ def create_couchbase_worker(self, replicas=1):
...
                 )
 
-    def run_workload(self, replicas):
+    def run_workload(self, replicas, run_in_bg=False):
         """
         Running workload with pillow fight operator
         Args:
          replicas (int): Number of pods
-
+         run_in_bg (bool) : Optional run IOs in background
```
Comment (ebenahar) [inline ocs_ci/ocs/couchbase.py:216]: missing indentation

File: ocs_ci/ocs/couchbase.py [hunk 5]
```diff
@@ -203,23 +208,48 @@ def create_couchbase_worker(self, replicas=1):
...
+            executor = ThreadPoolExecutor(1)
+            self.result = executor.submit(PillowFight.run_pillowfights, self, replicas=replicas)
+            return self.result
         PillowFight.run_pillowfights(self, replicas=replicas)
 
-    def analyze_run(self):
+    def analyze_run(self, skip_analyze=False):
         """
         Analyzing the workload run logs
+        Args:
```
Comment (ebenahar) [inline ocs_ci/ocs/couchbase.py:229]: add an empty line above

File: tests/e2e/workloads/couchbase/test_couchbase_pod_respin.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        log.info(f"Respin Ceph pod {pod_name}")
+
+        if pod_name == 'couchbase':
+            self.cb.respin_couchbase_app_pod()
+        else:
+            disruption = Disruptions()
+            disruption.set_resource(resource=f'{pod_name}')
+            disruption.delete_resource()
+
+        while not (self.cb.result.done()):
```
Comment (ebenahar) [inline tests/e2e/workloads/couchbase/test_couchbase_pod_respin.py:54]: can this end up running endlessly?

Comment (shreekarSS) [reply tests/e2e/workloads/couchbase/test_couchbase_pod_respin.py]: Fixed now .Thanks

File: tests/e2e/workloads/couchbase/test_couchbase_pod_respin.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+                marks=pytest.mark.polarion_id("OCS-779")),
+            pytest.param(
+                *['mgr'],
+                marks=pytest.mark.polarion_id("OCS-781")),
+            pytest.param(
+                *['couchbase'],
+                marks=pytest.mark.polarion_id("OCS-786")),
+        ])
+    @pytest.mark.usefixtures(cb_setup.__name__)
+    def test_run_couchbase_respin_pod(self, pod_name):
```
Comment (ebenahar) [inline tests/e2e/workloads/couchbase/test_couchbase_pod_respin.py:45]: better to send the fixture as an arg to the test function

Comment (shreekarSS) [reply tests/e2e/workloads/couchbase/test_couchbase_pod_respin.py]: ack

**Pull request merged**